src/backend/src/utils/reimbursement-requests.utils.ts

```diff
@@ -50,3 +58,73 @@ export const validateReimbursementProducts = async (reimbursementProductCreateAr
     throw new HttpException(400, `The following projects or work packages do not exist: ${missingWbsNumbers.join(', ')}`);
   }
 };
+
+/**
+ * This function updates the current reimbursement products of a reimbursement request versus the new reimbursement products of a reimbursement request.
+ * @param currentReimbursementProducts the current reimbursement products of a reimbursement request
+ * @param updatedReimbursementProducts the new reimbursement products to compare
+ * @param reimbursementRequestId the reimbursement request that is being changed id
+ */
+export const updateReimbursementProducts = async (
+  currentReimbursementProducts: Reimbursement_Product[],
+  updatedReimbursementProducts: ReimbursementProductCreateArgs[],
+  reimbursementRequestId: string
+) => {
+  //if a product has an id that means it existed before and was updated
+  const updatedProducts = updatedReimbursementProducts.filter((product) => product.id);
+
+  //Check to make sure that the updated products actually exist in the database
+  const prismaProductIds = currentReimbursementProducts.map((product) => product.reimbursementProductId);
+
+  const missingProductIds = updatedProducts.filter((product) => !prismaProductIds.includes(product.id!));
+
+  if (missingProductIds.length > 0) {
+    throw new HttpException(
+      400,
+      `The following products do not exist: ${missingProductIds.map((product) => product.name).join(', ')}`
+    );
+  }
+
+  const updatedProductIds = updatedProducts.map((product) => product.id!);
+
+  //if the product does not have an id that means it is new
+  const newProducts = updatedReimbursementProducts.filter((product) => !product.id);
+
+  //if there are products that exist in the current database but were not included in this edit, that means they were deleted
+  const deletedProducts = currentReimbursementProducts.filter(
+    (product) => !updatedProductIds.includes(product.reimbursementProductId)
+  );
+
+  //update the deleted reimbursement products by setting their date deleted to now
```

Comment:
each of these prisma calls can be helper functions that just take in the things they need. It'll help with readability and separate out the logic a little.